android-core/src/main/kotlin/com/mparticle/internal/ProviderPersistence.kt

```diff
@@ -0,0 +1,132 @@
+package com.mparticle.internal
+
+import android.content.Context
+import com.mparticle.internal.MPUtility.isEmpty
+import org.json.JSONObject
+import java.security.SecureRandom
+import java.util.Locale
+import java.util.UUID
+import java.util.regex.Pattern
+
+/**
+ * This class is responsible for pulling persistence from files from *other* SDKs, and serializing itself as a part of a batch.
+ * The idea here is that a customer may want to remove an SDK from their app and move it
+ * server side via mParticle. Rather than start from stratch, it's crucial that we can query data that
+ * the given SDK had been storing client-side.
+ */
+internal class ProviderPersistence(config: JSONObject, context: Context) : JSONObject() {
+    init {
+        val configPersistence = config.getJSONArray(KEY_PERSISTENCE)
+        for (i in 0 until configPersistence.length()) {
+            val values = JSONObject()
+            if (configPersistence.getJSONObject(i).has(KEY_PERSISTENCE_ANDROID)) {
+                val files = configPersistence.getJSONObject(i).getJSONArray(KEY_PERSISTENCE_ANDROID)
+
+                for (fileIndex in 0 until files.length()) {
+                    val fileObject = files.getJSONObject(fileIndex)
+                    val preferences =
+                        context.getSharedPreferences(fileObject.getString(KEY_PERSISTENCE_FILE), fileObject.getInt(KEY_PERSISTENCE_MODE))
+                    val fileObjects = fileObject.getJSONArray(KEY_PERSISTENCE_KEY_LIST)
+                    val editor = preferences.edit()
+                    for (keyIndex in 0 until fileObjects.length()) {
+                        val type = fileObjects.getJSONObject(keyIndex).getInt(KEY_PERSISTENCE_TYPE)
+                        val key = fileObjects.getJSONObject(keyIndex).getString(KEY_PERSISTENCE_KEY)
+                        val mpKey = fileObjects.getJSONObject(keyIndex).getString(KEY_PERSISTENCE_MPVAR)
```

Comment:
i suggest turn `fileObjects.getJSONObject(keyIndex)` into a variable for readability